c10/core/impl/SizesAndStrides.h

```diff
@@ -17,26 +30,118 @@ class C10_API SizesAndStrides {
   using strides_iterator = int64_t*;
   using strides_const_iterator = const int64_t*;
 
-  SizesAndStrides() : sizes_{0}, strides_{1} {}
+  SizesAndStrides() {
+    setInlineSize(1);
+    size_at_unchecked(0) = 0;
+    stride_at_unchecked(0) = 1;
+  }
+
+  ~SizesAndStrides() {
+    if (C10_UNLIKELY(!isInline())) {
+      delete outOfLineStorage();
+    }
+  }
+
+  SizesAndStrides(const SizesAndStrides& rhs) {
+    if (C10_LIKELY(rhs.isInline())) {
+      copyFromInline(rhs);
+    } else {
+      taggedStorageOrSize_ = reinterpret_cast<int64_t>(new std::vector<int64_t>(*rhs.outOfLineStorage()));
+    }
+  }
+
+  SizesAndStrides& operator=(const SizesAndStrides& rhs) {
+    if (this == &rhs) {
+      return *this;
+    }
+    if (C10_LIKELY(rhs.isInline())) {
+      if (C10_UNLIKELY(!isInline())) {
+        delete outOfLineStorage();
```

Comment:
This is all hella manual but I guess we don't have a choice if we're going to go the tagged pointer route.